AlphaWallet/Core/Formatters/CurrencyFormatter.swift

```diff
@@ -2,47 +2,62 @@
 
 import Foundation
 
-class CurrencyFormatter {
-    static var formatter: NumberFormatter {
-        let formatter = NumberFormatter()
-        formatter.minimumFractionDigits = 2
-        formatter.maximumFractionDigits = 2
-        formatter.roundingMode = .down
-        //TODO support multiple currency values
-        formatter.currencyCode = Currency.USD.rawValue
-        formatter.numberStyle = .currency
+extension NumberFormatter {
 
-        return formatter
+    static let currency = Formatter(.currency)
+    static let usd = Formatter(.usd)
+    static let short = Formatter(.short)
+
+    class Formatter {
+        private let formatter: NumberFormatter
+
+        fileprivate init(_ configuration: NumberFormatterConfiguration) {
+            formatter = configuration.formatter
+        }
+
+        func string(from number: NSNumber) -> String? {
+            return formatter.string(from: number)
+        }
+
+        func string(for: Any?) -> String? {
```

Comment:
But I realise we have both:

```
func string(from number: NSNumber) -> String? {
    return formatter.string(from: number)
}

func string(for: Any?) -> String? {
    return formatter.string(for: number)
}
```

It seems like `NumberFormatter.string(from:)` and `Formatter.string(for:)` do the same thing. Do you know if they are different? Otherwise let's just use `NumberFormatter.string(from:)` since it's in a subclass.

It also looks like we always call with a `Double`, so we can remove this:

```
func string(from number: NSNumber) -> String? {
```

and replace the second version with this:

```
func string(from number: Double) -> String? {
    return formatter.string(from: number as NSNumber)
}
```

What do you think?

Reply:
yep it will be great improvement, i will update it.